File: leetcode/num75.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        red = 0
        blue = len(nums)-1
        for i in range(0, blue+1):
            if nums[i] == 0:
                print('change i and red', i, red)
                tmp = nums[i]
                nums[i] = nums[red]
                nums[red] = tmp
                red += 1
                
            elif nums[i] == 2:
                print('change i and blue', i, blue)
                tmp = nums[i]
                nums[i] = nums[blue]
                nums[blue] = tmp
                i = i-1
                blue -= 1
                
            print(red, blue, nums)

if __name__ == "__main__":
    nums = [2,0,2,1,1,0]
    solution = Solution()
    solution.sortColors(nums)
    # print(nums)

File: leetcode/num75.java
package leetcode;

class num75 {
    public void sortColors(int[] nums) {
        int red = 0;
        int blue = nums.length - 1;
        for (int i = 0; i <= blue; i++) {
            if (nums[i] == 0) {
                int tmp = nums[i];
                nums[i] = nums[red];
                nums[red] = tmp;
                red += 1;
            } else if (nums[i] == 2) {
                int tmp = nums[i];
                nums[i] = nums[blue];
                nums[blue] = tmp;
                i -= 1;
                blue -= 1;
            }
        }
    }

    public static void main(String[] args) {
        int[] nums = { 2, 0, 2, 1, 1, 0 };
        num75 solution = new num75();
        solution.sortColors(nums);
        for (int item : nums) {
            System.out.println(item);
        }
    }
}